lib/ConstExtract/ConstExtract.cpp

```diff
@@ -489,14 +500,45 @@ void writeAttributes(
   });
 }
 
+void writeResultBuilderInformation(llvm::json::OStream &JSON,
+                                   const swift::NominalTypeDecl *TypeDecl,
+                                   const swift::VarDecl *VarDecl) {
+  if (auto *attr = VarDecl->getAttachedResultBuilder()) {
+    JSON.attributeObject("resultBuilder", [&] {
+      JSON.attribute("type", toFullyQualifiedTypeNameString(attr->getType()));
+      writeFileInformation(JSON, VarDecl->getASTContext(), attr);
+    });
+
+    return;
+  }
+
+  for (ProtocolDecl *Decl :
+       TypeDecl->getLocalProtocols(ConformanceLookupKind::All)) {
+    for (auto Member : Decl->getMembers()) {
+      if (auto *VD = dyn_cast<swift::VarDecl>(Member)) {
+        if (auto *attr = VD->getAttachedResultBuilder()) {
+          if (VD->getName() == VarDecl->getName()) {
+            JSON.attributeObject("resultBuilder", [&] {
+              JSON.attribute("type",
+                             toFullyQualifiedTypeNameString(attr->getType()));
+              writeFileInformation(JSON, VD->getASTContext(), attr);
+            });
```

Comment:
I likely want to break out of this loop when we see a result builder.

Reply:
Should be sufficient to break once we've seen a var decl with the name to the one we're looking for, result builder or no result builder, right?

Reply:
Yes. I think so.